crates/ruff_macros/src/newtype_index.rs

```diff
@@ -45,6 +45,9 @@ pub(super) fn generate_newtype_index(item: ItemStruct) -> syn::Result<proc_macro
                 // SAFETY:
                 // * The `value < u32::MAX` guarantees that the add doesn't overflow.
                 // * The `+ 1` guarantees that the index is not zero
+                //
+                // N.B. We have to use the unchecked variant here because we're
+                // in a const context and Option::unwrap isn't const yet.
```

Comment:
Is this ever called in a const context? We had some `const fn` that didn't actually need to be const

Reply:
It is yeah: https://github.com/astral-sh/ruff/blob/995649f45584af2dc8a1d128bfb1ed852dfcaf99/crates/ruff_macros/src/newtype_index.rs#L40

Anyway you slice it, `Self::MAX` having type `Self` means you need to build a `NonZeroU32` in a const context. And to do that for `u32::MAX - 1`, I believe that requires some kind of `unsafe` uttered somewhere. It's possible that `Self::MAX` doesn't need to be const and could be a method instead, but that's a bigger change.

I think we can probably just leave this one here for now.